interfaces/cython/cantera/reactor.pyx

```diff
@@ -23,8 +23,20 @@ cdef class ReactorBase:
     Common base class for reactors and reservoirs.
     """
     reactor_type = "none"
-    def __cinit__(self, *args, **kwargs):
-        self._reactor = newReactor(stringify(self.reactor_type))
+    def __cinit__(self, _SolutionBase contents=None, name=None, *, **kwargs):
+        def reactor_name(name):
+            if name is not None:
+                return name
+            _reactor_counts[self.reactor_type] += 1
+            return f"{self.reactor_type}_{_reactor_counts[self.reactor_type]}"
+
+        if isinstance(contents, _SolutionBase):
+            self._reactor = newReactor(stringify(self.reactor_type),
+                                       contents._base, stringify(reactor_name(name)))
+        else:
+            # deprecated: will raise warnings in C++ layer
+            self._reactor = newReactor(stringify(self.reactor_type))
+            self._reactor.get().setName(stringify(reactor_name(name)))
```

Comment:
Is the second branch here (and having a default value for `contents` even needed? In Cantera 3.0, calling `ct.Reactor()` already issues the following warning:
```
DeprecationWarning: Starting in Cantera 3.1, the reactor contents must not be empty.
  r = ct.Reactor()
```
So I think at least for the Python interface, it's acceptable to remove the option not to specify the contents as part of the constructor.

Reply:
We introduced this requirement in 3.1 via #1663, i.e. the warning should be updated.